content/docs/2_cookbook/9_setup/0_caddy/cookbook-recipe.txt

```diff
@@ -0,0 +1,124 @@
+Title: Running Kirby on a Caddy v2 web server
+
+----
+
+Published: 2023-08-03
+
+----
+
+Description: How to configure the popular and simple Caddy v2 web server to run a Kirby website.
+
+----
+
+Authors:
+
+- authors/moritz-stueckler
+
+----
+
+Text:
+
+[Kirby states](https://getkirby.com/docs/guide/quickstart) that it is able to run on many different web servers. However, in reality it is mostly Apache and sometimes Nginx which are being used as webservers to run Kirby on. But there is also another, modern and very easy-to-configure alternative in the webserver world: [Caddy v2](https://caddyserver.com/v2).
+
+## Caddy: No support for `.htaccess`
+
+Just like Nginx the Caddy webserver does not use the famous and project-specific `.htaccess` files that Kirby ships with. So instead of using this file, you'll have to configure the webserver manually through a special configuration file called `Caddyfile`. This file should typically be placed next to the Caddy executable, or its location can be passed in, when Caddy is called.
+
+
+## Configuration via Caddyfile
+Caddy uses its own [configuration file format and syntax](https://caddyserver.com/docs/caddyfile/concepts). These Caddyfiles are typically very short and efficient and allow you to configure a webserver with very little content. This is also due to the fact that Caddy uses a lot of very good and modern defaults compared to older webservers like Apache. Therefore Caddy makes ist very easy to host websites with very little configuration and something like TLS only requires a single line in a Caddy configuration.
+
+<info>
+Web server configuration is a very big topic and there are many things to consider for every individual use case. We can only offer a **starting point** for a good config and we can't cover every aspect in this article. So this article will not cover every aspect of webserver configuration (e.g. performance-related optimizations).
+</info>
+
+## Boilerplate config
+
+```
+(common) {
+	php_fastcgi php:9000 # Adjust to your setup
+	tls name@user.com # Adjust to your setup
+	file_server
+}
+
+(kirby) {
+	@blocked {
+		path *.txt *.md /content/* /site/* /kirby/* /.*
```

Comment:
Why the matchers for `*.txt` and `*.md`? Shouldn't they already be covered by the `/content/*` matcher?

If possible, it would be great to allow access to the `.well-known` folder like in the `.htaccess`. But since Caddy handles Let's Encrypt, it's probably not as critical.

Reply:
- The `*.txt` and `*.md` are basically for any misplaced files which mistakenly ended up outside of the content directory.
- Allowing `.well-known` while disallowing other hidden files would require a `path_regexp` matcher, which makes the whole Caddyfile a lot harder to understand imo. Since Caddy handles TLS automatically, the amount of use cases where `.well-known` is used for anything other than Let's encrypt is pretty mall in my opinion.